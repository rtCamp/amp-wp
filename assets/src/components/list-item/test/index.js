/**
 * External dependencies
 */
import { create } from 'react-test-renderer';

/**
 * Internal dependencies
 */
import { ListItem } from '..';

describe( 'ListItem', () => {
	it( 'matches snapshots', () => {
		// Normal listing.
		let wrapper = create(
			<ListItem
				items={
					[
						{ value: 'Item 1' },
						{ value: 'Item 2' },
					]
				}
			/>,
		);

		expect( wrapper.toJSON() ).toMatchSnapshot();

		// Listing with keys.
		wrapper = create(
			<ListItem
				items={
					[
						{ label: 'Label 1', value: 'Item 1' },
						{ label: 'Label 2', value: 'Item 2' },
					]
				}
			/>,
		);

		expect( wrapper.toJSON() ).toMatchSnapshot();

		// Listing with hading.
		wrapper = create(
			<ListItem
				heading="List heading"
				items={
					[
						{ label: 'Label 1', value: 'Item 1' },
						{ label: 'Label 2', value: 'Item 2' },
					]
				}
			/>,
		);

		expect( wrapper.toJSON() ).toMatchSnapshot();
	} );
} );
