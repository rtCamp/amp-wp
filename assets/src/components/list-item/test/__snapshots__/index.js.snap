// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem matches snapshots 1`] = `
<ul
  className="list-items"
>
  <li
    className="list-items__item"
  >
    <span
      className="list-items__item-value"
    >
      Item 1
    </span>
  </li>
  <li
    className="list-items__item"
  >
    <span
      className="list-items__item-value"
    >
      Item 2
    </span>
  </li>
</ul>
`;

exports[`ListItem matches snapshots 2`] = `
<ul
  className="list-items"
>
  <li
    className="list-items__item"
  >
    <strong
      className="list-items__item-key"
    >
      Label 1
    </strong>
    <span
      className="list-items__item-value"
    >
      Item 1
    </span>
  </li>
  <li
    className="list-items__item"
  >
    <strong
      className="list-items__item-key"
    >
      Label 2
    </strong>
    <span
      className="list-items__item-value"
    >
      Item 2
    </span>
  </li>
</ul>
`;

exports[`ListItem matches snapshots 3`] = `
<ul
  className="list-items"
>
  <li
    className="list-items__item"
  >
    <h4
      className="list-items__heading"
    >
      List heading
    </h4>
  </li>
  <li
    className="list-items__item"
  >
    <strong
      className="list-items__item-key"
    >
      Label 1
    </strong>
    <span
      className="list-items__item-value"
    >
      Item 1
    </span>
  </li>
  <li
    className="list-items__item"
  >
    <strong
      className="list-items__item-key"
    >
      Label 2
    </strong>
    <span
      className="list-items__item-value"
    >
      Item 2
    </span>
  </li>
</ul>
`;
